pkg/resourcemanager/controller/secret/reconciler.go

```diff
@@ -66,6 +68,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			// check if we are responsible for this MR, class might have changed, then we need to remove our finalizer
 			if ref.Name == secret.Name && r.ClassFilter.Responsible(&resource) {
 				secretIsReferenced = true
+				if _, ok := secret.Labels[v1alpha1.ReferencedBy]; !ok {
```

Comment:
You can also add a statement in `docs/concepts/resource-manager.md` that the Secret controller is now adding this label.